POOSDKer/Backend/NetworkingController.swift

```diff
@@ -6,105 +6,263 @@
+        self.serviceAdvertiser = MCNearbyServiceAdvertiser(peer: appState.peerID, discoveryInfo: discoveryInfo, serviceType: serviceType)
+        self.serviceAdvertiser.delegate = self
+        appState.connectedPeers = []
+        appState.connectedPeers.append(Peer(id: appState.UID, displayName: appState.settings.displayName, playerColor: appState.settings.playerColor, mcPeerID:appState.peerID))
+        appState.hostPeer = appState.connectedPeers[0]
+        serviceAdvertiser.startAdvertisingPeer()
+        
+        
+     
+    }
+    
+    func stopHosting() {
+        print("Stopping hosting...")
+        serviceAdvertiser.stopAdvertisingPeer()
+        appState.hostPeer = nil
+    }
+    
+    func startBrowsing() {
+        print("Starting browsing...")
+        appState.discoveredPeers = []
+        serviceBrowser.startBrowsingForPeers()
+    }
+    
+    func stopBrowsing() {
+        print("Stopping browsing...")
+        serviceBrowser.stopBrowsingForPeers()
+    }
+    
+    
+    func requestToJoinHost(hostPeer: Peer) {
+        let infoToSend = ["displayName": appState.settings.displayName, "id": appState.UID, "playerColor": appState.settings.playerColor]
+        if let peerID = hostPeer.mcPeerID {
+            if let context = try? JSONEncoder().encode(infoToSend) {
+                serviceBrowser.invitePeer(peerID, to: mcSession, withContext: context, timeout: 30)
+                appState.hostPeer = hostPeer;
             }
-
-            listener?.newConnectionHandler = { connection in
-                // MARK: Handle the connection with a response for the websocket in the future
-                connection.start(queue: .main)
+        }
+        else {
+            print("Attempting to connect to host without knowledge of mcPeerID")
+        }
+    }
+    
+    
+    func disconnectFromHost() {
+        mcSession.disconnect()
+        appState.hostPeer = nil;
+    }
+    
+    func advertiser(_ advertiser: MCNearbyServiceAdvertiser, didReceiveInvitationFromPeer peerID: MCPeerID, withContext context: Data?, invitationHandler: @escaping (Bool, MCSession?) -> Void) {
+        if let context = context, let receivedInfo = try? JSONDecoder().decode([String: String].self, from: context) {
+            // Now you have the additional info
+            if let displayName = receivedInfo["displayName"], let uniqueID = receivedInfo["id"], let playerColor = receivedInfo["playerColor"] {
+                self.appState.connectedPeers.append(Peer(id: uniqueID, displayName: displayName, playerColor: playerColor, mcPeerID: peerID))
+                print("Connecting user with name: " + displayName)
             }
-
-            listener?.start(queue: .main)
+        }
+        
+        // You can decide whether to accept the invitation based on the received information
+        invitationHandler(true, mcSession)
+    }
+    
+    
+    // !! It may make sense in the future to abstract out the process of broadcasting
```

Comment:
Do the exclamation marks have a specific meaning? Maybe we should settle on a specific "TO-DO", "URGENT", "FUT. TASK", etc. standard to mark sections.

Reply:
Xcode has different ways to markup code with comments. !! is one. There is also MARK: that created a mark

Reply:
Is there any documentation you can link for this? Would be helpful

Reply:
https://stackoverflow.com/questions/7600435/what-source-comments-does-xcode-recognize-as-tags


There are quite a few not documented but this thread has a lot of em. Ofc theres no offical docs 🙃